addons/report/models/report.py

```diff
@@ -73,50 +69,40 @@ def _get_wkhtmltopdf_bin():
         wkhtmltopdf_state = 'workers'
 
 
-class Report(osv.Model):
+class Report(models.Model):
     _name = "report"
     _description = "Report"
 
     public_user = None
 
-    #--------------------------------------------------------------------------
-    # Extension of ir_ui_view.render with arguments frequently used in reports
-    #--------------------------------------------------------------------------
-
-    def render(self, cr, uid, ids, template, values=None, context=None):
+    @api.multi
+    def render(self, template, values=None):
         """Allow to render a QWeb template python-side. This function returns the 'ir.ui.view'
         render but embellish it with some variables/methods used in reports.
 
         :param values: additionnal methods/variables used in the rendering
         :returns: html representation of the template
         """
-        if values is None:
+        if not values:
             values = {}
 
-        if context is None:
-            context = {}
-
-        context = dict(context, inherit_branding=True)  # Tell QWeb to brand the generated html
-
-        view_obj = self.pool['ir.ui.view']
-
-        user = self.pool['res.users'].browse(cr, uid, uid, context=context)
+        context = dict(self.env.context, inherit_branding=True)  # Tell QWeb to brand the generated html
+        user = self.env['res.users'].browse(self.env.uid)
         website = None
         if request and hasattr(request, 'website'):
             if request.website is not None:
                 website = request.website
-                context = dict(context, translatable=context.get('lang') != request.website.default_lang_code)
-
+                context.update({'translatable': context.get('lang') != request.website.default_lang_code})
         values.update(
             time=time,
-            context_timestamp=lambda t: fields.datetime.context_timestamp(cr, uid, t, context),
+            context_timestamp=lambda t: fields.Datetime.context_timestamp(self.with_context(tz=user.tz), t),
```

Comment:
Because of this. In internal header we used user timezone and based on that we print current native User date and time..
see, 
https://github.com/odoo/odoo/blob/master/addons/report/views/layouts.xml#L164